src/point_one/fusion_engine/messages/configuration.h

```diff
@@ -391,6 +391,356 @@ struct alignas(4) CoarseOrientation {
+  uint16_t num_msgs = 0;
+  /** Placeholder pointer for variable length set of messages. */
+  MsgRate msg_rates[0];
+};
+
+/**
+ * @brief Configuration for the streams associated with an output interface.
+ *
+ * This message is followed by `N` `uint8_t` stream indices, where `N`
+ * is equal to @ref num_streams. For example:
+ *
+ * ```
+ * {MessageHeader, SetConfigMessage, OutputInterfaceConfig,
+ *  uint8_t, uint8_t,  ...}
+ * ```
+ */
+struct alignas(4) OutputInterfaceConfig {
+  /** The output interface to configure. */
+  InterfaceID output_interface;
+  /** The number of `stream_indices` entries this message contains. */
+  uint8_t num_streams = 0;
+  uint8_t reserved[3] = {0};
+  /**
+   * Placeholder pointer for variable length set of indices.
+   *
+   * In the future these streams will be user defined, but for now they are as:
+   * - `0`: All FusionEngine messages.
+   * - `1`: All NMEA messages.
+   * - `2`: All RTCM messages.
+   */
+  uint8_t stream_indices[0];
+};
+
+/**
+ * @brief Configure the set of output streams enabled for a given output
+ *        interface. (@ref MessageType::SET_OUTPUT_INFERFACE_STREAMS, version
+ *        1.0).
+ * @ingroup config_and_ctrl_messages
+
+ * The device will respond with a @ref CommandResponseMessage indicating whether
+ * or not the request was accepted. Not all interfaces defined in @ref
+ * InterfaceID are supported on all devices.
+ *
+ * Parameter changes are applied to the device's active configuration
+ * immediately, but are not saved to persistent storage and will be restored to
+ * their previous values on reset. To save configuration settings to persistent
+ * storage, see @ref SaveConfigMessage.
+ */
+struct alignas(4) SetOutputInterfaceStreamsMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE =
+      MessageType::SET_OUTPUT_INFERFACE_STREAMS;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+  /**
+   * The type of action this configuration message applies to the
+   * previous list of streams.
+   */
+  UpdateAction update_action = UpdateAction::REPLACE;
+  uint8_t reserved[3] = {0};
+
+  /**
+   * The configuration for an output interface to apply.
+   */
+  OutputInterfaceConfig output_interface_data;
+};
+
+/**
+ * @brief Query the sets of output streams enabled each output
+ *        interface (@ref
+ *        MessageType::GET_OUTPUT_INFERFACE_STREAMS, version 1.0).
+ * @ingroup config_and_ctrl_messages
+ *
+ * The device will respond with a @ref OutputInterfaceStreamsDataMessage
+ * containing the values.
+ */
+struct alignas(4) GetOutputInterfaceStreamsMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE =
+      MessageType::GET_OUTPUT_INFERFACE_STREAMS;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+
+  /** The config source to request data from (active, saved, etc.). */
+  ConfigurationSource request_source = ConfigurationSource::ACTIVE;
+
+  uint8_t reserved[3] = {0};
```

Comment:
So to be clear, a set is for a specific interface but a get gives you all of them? Not sure I like that being asymmetrical. Is there a way we could include the specific interface in the get, and maybe include an option to return all?